wolfcrypt/src/asn.c

```diff
@@ -33283,6 +33341,19 @@ static int ParseCRL_Extensions(DecodedCRL* dcrl, const byte* buf,
             }
         #endif
         }
+        else if (oid == CRL_NUMBER_OID) {
+            localIdx = idx;
+            if (GetASNTag(buf, &localIdx, &tag, sz) == 0 && tag == ASN_INTEGER) {
```

Comment:
this needs to handle numbers larger than one byte, maybe something like the following:

```
@@ -33000,6 +33058,42 @@ static int ParseCRL_Extensions(DecodedCRL* dcrl, const byte* buf,
             }
         #endif
         }
+        else if (oid == CRL_NUMBER_OID) {
+            localIdx = idx;
+            if (GetASNTag(buf, &localIdx, &tag, sz) == 0 && tag == ASN_INTEGER) {
+                ret = GetASNInt(buf, &idx, &length, sz);
+                if (ret < 0) {
+                    WOLFSSL_MSG("\tcouldn't parse CRL number extension");
+                    return ret;
+                }
+                else {
+                    if (length > 1) {
+                        mp_int m;
+                        int    i;
+
+                        mp_init(&m);
+                        ret = mp_read_unsigned_bin(&m, buf + idx, length);
+                        if (ret != MP_OKAY) {
+                            mp_free(&m);
+                            return BUFFER_E;
+                        }
+
+                        dcrl->crlNumber = 0;
+                        for (i = 0; i < m.used; ++i) {
+                            if (i > (int)sizeof(word32)) {
+                                    break;
+                            }
+                            dcrl->crlNumber |= ((word32)m.dp[i]) <<
+                                (DIGIT_BIT * i);
+                        }
+                        mp_free(&m);
+                    }
+                    else {
+                        dcrl->crlNumber = buf[idx];
+                    }
+                }
+            }
+        }
 
         idx += length;
     }
@@ -33267,6 +33361,8 @@ end:
```

Reply:
Added